frameworklauncher/src/main/java/com/microsoft/frameworklauncher/common/web/WebClient.java

```diff
@@ -41,10 +43,15 @@
   private CloseableHttpClient httpClient;
   private String baseURI;
 
-  public WebClient(String baseURI, String launchClientType) {
+  public WebClient(String baseURI, LaunchClientType launchClientType, String userName, Boolean isAdmin) {
     this.baseURI = baseURI;
-    Header header = new BasicHeader(WebCommon.REQUEST_HEADER_LAUNCH_CLIENT_TYPE, launchClientType);
-    this.httpClient = HttpClients.custom().setDefaultHeaders(Arrays.asList(header)).build();
+
+    List<Header> headers = new ArrayList<>();
+    headers.add(new BasicHeader(WebCommon.REQUEST_HEADER_LAUNCH_CLIENT_TYPE, launchClientType.toString()));
+    headers.add(new BasicHeader(WebCommon.REQUEST_HEADER_USER_NAME, userName));
+    headers.add(new BasicHeader(WebCommon.REQUEST_HEADER_IS_ADMIN, isAdmin.toString()));
```

Comment:
>headers.add(new BasicHeader(WebCommon.REQUEST_HEADER_IS_ADMIN, isAdmin.toString())); [](start = 4, length = 84)

I would think as we already specified users in ACL configuration, we should also specifies admins in ACL configuration file.

Reply:
For simple design in first stage, we trust external. And if he specifies IsAdmin = true, he can even do not provide the UserName. 
BTW, in this way, we do not need to store Admin info and Admin only one call to request instead of first add one Admin then request.

---
In reply to: [167463957](https://github.com/Microsoft/pai/pull/320#discussion_r167463957) [](ancestors = 167463957)